### PR TITLE
feat: Create repository-level Coverage pages COV-256

### PR DESCRIPTION
**Pull request by prcr**
Adds repository-level pages specific to the Coverage product.

### :eyes: Live preview
- https://feat-create-coverage-pages-cov-256--docs-codacy.netlify.app/coverage/repositories/repository-dashboard/
- https://feat-create-coverage-pages-cov-256--docs-codacy.netlify.app/coverage/repositories/commits/
- https://feat-create-coverage-pages-cov-256--docs-codacy.netlify.app/coverage/repositories/files/
- https://feat-create-coverage-pages-cov-256--docs-codacy.netlify.app/coverage/repositories/pull-requests/

### :construction: To do
-   [x] Review and fix TODOs (I'll update the screenshots on a separate pull request)

### CODE REVIEW

File: docs/coverage/repositories/repository-dashboard.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+-   A green or red indicator depending if coverage is within the acceptable level or not
+-   The current coverage value
+-   The coverage variation introduced by the last commit
+
+!!! note
+    The chart only displays a value if Codacy received coverage data for the most recent commit. This is because one commit can <span class="skip-vale">easily</span> change the size or number of files on the repository, or even remove some files that had coverage information.
+
+The chart also displays the **trendline** based on the past behavior and the **quality goal** defined on the repository [quality settings](../../repositories-configure/adjusting-quality-settings.md).
+
+![Quality evolution chart](images/repository-dashboard-quality-evolution.png)<!--TODO Update-->
```
Comment (nicklem) [inline docs/coverage/repositories/repository-dashboard.md:46]: Another inverted image, where the original page and the Coverage page point to each other's image in the built files (I see no changes in the source Markdown of the original page).

Seems like a build process issue tied to the relative paths of the assets.

Could this be an mkdocs bug?

File: docs/coverage/repositories/repository-dashboard.md [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+The tab displays the following information:
+
+-   A green or red indicator depending if coverage is within the acceptable level or not
+-   The current coverage value
+-   The coverage variation introduced by the last commit
+
+!!! note
+    The chart only displays a value if Codacy received coverage data for the most recent commit. This is because one commit can <span class="skip-vale">easily</span> change the size or number of files on the repository, or even remove some files that had coverage information.
+
+The chart also displays the **trendline** based on the past behavior and the **quality goal** defined on the repository [quality settings](../../repositories-configure/adjusting-quality-settings.md).
```
Comment (nicklem) [inline docs/coverage/repositories/repository-dashboard.md:44]: * Should this be "coverage goal" now?
* I'm not sure if these are still "quality settings" or just "settings" at this point (seeing it elsewhere):

```suggestion
The chart also displays the **trendline** based on the past behavior and the **coverage goal** defined on the repository [settings](../../repositories-configure/adjusting-quality-settings.md).
```

File: docs/coverage/repositories/repository-dashboard.md [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    Review page, see https://codacy.atlassian.net/browse/COV-115 for more details-->
+
+The **Repository Dashboard** provides an overview of the repository coverage and items that require your attention.
+
+To access your Repository Dashboard, select a repository from the [Organization Dashboard](../../organizations/organization-overview.md) or open a repository on any other page and select **Dashboard** on the left navigation sidebar.<!--TODO Review navigation-->
+
+!!! tip
+    You can share the URL of the Repository Dashboard for your **public repositories** to allow other people to see your repository coverage metrics, even if they aren't registered on Codacy.
+
+![Repository Dashboard](images/repository-dashboard.png)<!--TODO Update-->
```
Comment (nicklem) [inline docs/coverage/repositories/repository-dashboard.md:16]: This image and the one on the [equivalent Quality page](https://feat-create-coverage-pages-cov-256--docs-codacy.netlify.app/repositories/repository-dashboard/#repository-dashboard) seem inverted here.

File: docs/coverage/repositories/commits.md [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+![{{ page.meta.page_name.capitalize() }} quality overview](images/{{ page.meta.file_name }}-detail-coverage-overview.png)<!--TODO Update-->
+<!-- vale on -->
+<!--coverage-overview-end-->
+
+<!--tab-files-start-->
+## Files tab
+
+The **Files** tab displays the coverage variation that the {{ page.meta.page_name }} introduced to the files in your repository relative to the {% if page.meta.page_name == "commit" %}parent commit{% else %}target branch{% endif %}, displayed either as a **positive or negative variation**, or **no variation** (represented by `=`):
+
+The option **Show also files without coverage changes** allows you to list all files that the {{ page.meta.page_name }} updated, even if their coverage didn't change.
```
Comment (nicklem) [inline docs/coverage/repositories/commits.md:74]: Nitpicks for present tense 😅 

```suggestion
The **Files** tab displays the coverage variation that the {{ page.meta.page_name }} introduces to the files in your repository relative to the {% if page.meta.page_name == "commit" %}parent commit{% else %}target branch{% endif %}, displayed either as a **positive or negative variation**, or **no variation** (represented by `=`):

The option **Show also files without coverage changes** allows you to list all files that the {{ page.meta.page_name }} updates, even if their coverage doesn't change.
```

File: docs/coverage/repositories/pull-requests.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+page_name: pull request
+file_name: pull-requests
+product: coverage
+---
+
+# [Coverage] Pull Requests page
+<!--TODO
+    Review page, see https://codacy.atlassian.net/browse/COV-119 for more details-->
+
+The **Pull Requests page** displays an overview of the pull requests in your repository, such as the analysis status and coverage metrics for each pull request. This allows you to monitor the coverage of the work in progress in your repository.
```
Comment (nicklem) [inline docs/coverage/repositories/pull-requests.md:11]: Just wondering: is it still an "analysis" for coverage?

File: docs/coverage/repositories/pull-requests.md [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+
+By default, the page lists open pull requests but you can click the **Closed** tab at the top of the list to display the closed pull requests.
+
+![Pull Requests page](images/pull-requests.png)<!--TODO Update-->
+
+Click a specific commit to see detailed coverage information for that pull request.
```
Comment (nicklem) [inline docs/coverage/repositories/pull-requests.md:17]: Typo:

```suggestion
Click a specific pull request to see detailed coverage information for that pull request.
```

**Review by nicklem — APPROVED** [review]

**Comment by nicklem** [comment]
Left some comments, but LGTM as a WIP :shipit:

File: docs/coverage/repositories/repository-dashboard.md (hunk 2 above)
Comment (prcr) [reply docs/coverage/repositories/repository-dashboard.md:44]: For now, the tab is still called "Quality Settings". :+1: 

I'm making this more generic by referring just to "goals":

```suggestion
The chart also displays the **trendline** based on the past behavior and goal defined on the repository [quality settings](../../repositories-configure/adjusting-quality-settings.md).
```

File: docs/coverage/repositories/repository-dashboard.md [hunk 7]
```diff
@@ -0,0 +1,79 @@
...
+
+-   [Quality evolution chart](#quality-evolution-chart)
+-   [Coverage](#coverage)
+-   [Open pull requests](#open-pull-requests)
+
+The following sections provide a detailed overview of each dashboard area.
+
+## Quality evolution chart
+
+The **Quality evolution** chart displays the evolution of the repository code quality regarding [code coverage](../../faq/code-analysis/which-metrics-does-codacy-calculate.md#code-coverage). Click on **Last 3 months**, **Last 31 days**, or **Last 7 days** to select the time interval of the historical data to display on the chart.
```
[the remark this answers is not in the document]
Comment (prcr) [reply docs/coverage/repositories/repository-dashboard.md:33]: Good catch! :ok_hand:

File: docs/coverage/repositories/pull-requests.md (hunk 6 above)
Comment (prcr) [reply docs/coverage/repositories/pull-requests.md:17]: Great catch! :detective:

File: docs/coverage/repositories/pull-requests.md (hunk 5 above)
Comment (prcr) [reply docs/coverage/repositories/pull-requests.md:11]: I'm also a bit unsure. Here's a preview of the pull request list:

image

I'm changing this to reference "Status" only.

File: docs/coverage/repositories/commits.md (hunk 4 above)
Comment (prcr) [reply docs/coverage/repositories/commits.md:74]: Thanks, indeed it reads much better! :raised_back_of_hand: 

I also opened a separate pull request to apply these changes to the original pages: https://github.com/codacy/docs/pull/1677

File: docs/coverage/repositories/repository-dashboard.md (hunk 1 above)
Comment (prcr) [reply docs/coverage/repositories/repository-dashboard.md]: Another great catch!

I had updated the wrong screenshot files earlier on in this commit: https://github.com/codacy/docs/pull/1658/commits/0e3a00792ab0e3c59a116fe4f79f872f4d034d59

I already reverted the changes to the original screenshots and updated the correct ones: https://github.com/codacy/docs/pull/1658/commits/efebfe243416df5ce1fa11a434a2411ca5a13092

File: docs/coverage/repositories/repository-dashboard.md (hunk 3 above)
Comment (prcr) [reply docs/coverage/repositories/repository-dashboard.md:16]: Fixed, see https://github.com/codacy/docs/pull/1658#discussion_r1164400047! :bow:

**Pull request merged**